src/Components/SmartComponents/HybridInventory/ImmutableDevicesTab/ImmutableDevices.test.js

```diff
@@ -171,4 +170,21 @@ describe('ImmutableDevices', () => {
         screen.getAllByText('This page is temporarily unavailable')
     ).toHaveLength(1);
   });
+
+  test('should display error state when user is not permitted', () => {
+    useRbac.mockReturnValue([[false], false]);
+    useSelector.mockReturnValue({});
+
+    render(
+      <ComponentWithContext
+        renderOptions={{ store: ReducerRegistry.getStore() }}
+      >
+        <ImmutableDevices />
+      </ComponentWithContext>
+    );
+
+    expect(
+        screen.getAllByText('You do not have permissions to view or manage Vulnerability')
+    ).toHaveLength(1);
```

Comment:
nit-pick: since you are expecting exactly one occurrence of this text, it is enough to use getByText: it will error if it finds none or more than 1. Also, I am a real fan of toBeVisible because it guarantees the component is 1) in the DOM, 2) visible in the viewport (which is not always a case if 1) is fulfilled)

```suggestion
        screen.getByText('You do not have permissions to view or manage Vulnerability')
    ).toBeVisible();
```